File: j2k/testData/fileOrElement/nullability/nullableInitializer.java
import org.jetbrains.annotations.Nullable;

public class PlainTrain {
    public String nullableString(int p) {
        return p > 0 ? "response" : null;
    }

    public String notNullString(int p) {
        return "response";
    }

    public Object nullableObj(int p) {
        return p > 0 ? "response" : null;
    }

    public final String nullableInitializerField = nullableString(3);
    public final String nullableInitializerFieldCast = (String) nullableObj(3);

    public void testProperty() {
        nullableInitializerField.charAt(0)
        nullableInitializerFieldCast.charAt(0)
    }

    public void testLocalVariable() {
        String nullableInitializerVal = nullableString(3);
        String nullableInitializerValCast = (String) nullableObj(3);

        nullableInitializerVal.charAt(0)
        nullableInitializerValCast.charAt(0)
    }

    public final String myNotNullField = notNullString(1);

    public String notNullInitializerFieldNullableUsage = myNotNullField;
    public String notNullInitializerFieldNotNullUsage = myNotNullField;

    public String nullInitializerFieldNullableUsage = null;
    public String nullInitializerFieldNotNullUsage = null;

    public void testNotNull(@Nullable Object obj) {
        if (true) {
            notNullInitializerFieldNullableUsage = (String) obj;
            notNullInitializerFieldNotNullUsage = "str";
        }
        else {
            nullInitializerFieldNullableUsage = (String) obj;
            nullInitializerFieldNotNullUsage = "str";
        }
    }
}

File: j2k/testData/fileOrElement/nullability/nullableInitializer.kt
class PlainTrain {
    fun nullableString(p: Int): String? {
        return if (p > 0) "response" else null
    }

    fun notNullString(p: Int): String {
        return "response"
    }

    fun nullableObj(p: Int): Any? {
        return if (p > 0) "response" else null
    }

    val nullableInitializerField = nullableString(3)
    val nullableInitializerFieldCast = nullableObj(3) as String?

    fun testProperty() {
        nullableInitializerField!![0]
        nullableInitializerFieldCast!![0]
    }

    fun testLocalVariable() {
        val nullableInitializerVal = nullableString(3)
        val nullableInitializerValCast = nullableObj(3) as String?

        nullableInitializerVal!![0]
        nullableInitializerValCast!![0]
    }

    val myNotNullField = notNullString(1)

    var notNullInitializerFieldNullableUsage: String? = myNotNullField
    var notNullInitializerFieldNotNullUsage = myNotNullField

    var nullInitializerFieldNullableUsage: String? = null
    var nullInitializerFieldNotNullUsage: String? = null

    fun testNotNull(obj: Any?) {
        if (true) {
            notNullInitializerFieldNullableUsage = obj as String?
            notNullInitializerFieldNotNullUsage = "str"
        } else {
            nullInitializerFieldNullableUsage = obj as String?
            nullInitializerFieldNotNullUsage = "str"
        }
    }
}